docs/notes-prove-verify.md

```diff
@@ -1,6 +1,392 @@
-In this module, the notes on how the steps 
-the prover must follow to construct
-PLONK proof, are documented.
+In this module, the notes on how the prover 
+algorithm is constructed
+for PLONK, is documented.
 
-Proving steps
-============
\ No newline at end of file
+PLONK proof construction 
+========================
+
+Following on from the generic
+SNARK construction, 
+here we will give the set up
+of a PLONK proof and show 
+which steps need to be satisfied
+to utilise the protocol.
+
+First we will explain the 
+derivation and simplification 
+of the arithmetic circuits. 
+
+PLONK uses both gate constraints 
+and copy constraints, to collect 
+like expressions. Using the same
+example of:
+
+\\[
+\mathbf{W}\_*L* \cdot \mathbf{W}\_*R* = \mathbf{W}\_*O*,
+\\]
+
+We can express multiples of the 
+same wires in or out of the same 
+gate, with the above equation.
+
+Thus we have 'ith' gates, so the 
+index from left or right across 
+the circuit is mitigated for 
+wires which are equal. 
+
+For example, in the two equations:
+
+\\[
+\begin{aligned}
+\mathbf{A}\_*1* \circ X \cdot \mathbf{B}\_*1* \circ \X^{2} 
+= \mathbf{C}\_*1*\\\\
+and 
+\mathbf{A}\_*2* \circ X^{2} \cdot \mathbf{B}\_*2* \circ \X 
+= \mathbf{C}\_*2*\\\\
+\end{aligned}
+\\]
+
+We can state the equalities that: 
+\\[
+\begin{aligned} 
+\mathbf{A}\_*1* = \mathbf{B}\_*2* \\\\
+&\\\\
+\mathbf{B}\_*1* = \mathbf{A}\_*2* \\\\
+\end{aligned}
+\\]
+
+These are examples of constraints 
+collected in PLONK. Which is done
+the same for addition gates, except
+the gate constrain satisfies:
+
+\\[
+\mathbf{W}\_*L* + \mathbf{W}\_*R* = \mathbf{W}\_*O*,
+\\]
+
+PLONK also uses 'copy
+constraints', which are used to 
+associate wires, which have 
+equality, from the entire circuit.
+These constriants are checked 
+with a permutation argument. 
+In essence, this checks that 
+wires are mot repeated by using 
+randomness given by the verifier. 
+
+This process is better explained in 
+the permutation [notes](notes-pa).
+
+After the constraints
+are made, they are formatted into a 
+system of mumerical equations, 
+which in PLONK are reduced to a 
+small amount of polynomial 
+equations which are capable of 
+representing the two types.
+PLONK allows us to combine the
+two gate equations by describing 
+their relationship relative to 
+the role in the circuit.
+PLONK also has constant, which
+are denoted as 'Q'. These 
+values will change for each 
+programme. When they are 
```

Comment:
We should mention here that they are actually defining the circuit "shape" and have nothing to do with the witness values

Reply:
Will add in.